src/Tempest/Core/src/PublishesFiles.php

```diff
@@ -5,65 +5,94 @@
 namespace Tempest\Core;
 
 use Closure;
-use Nette\InvalidStateException;
-use Tempest\Console\HasConsole;
 use Tempest\Generation\ClassManipulator;
-use function Tempest\src_namespace;
-use function Tempest\src_path;
+use Tempest\Generation\DataObjects\StubFile;
+use Tempest\Generation\Enums\StubFileType;
+use Tempest\Generation\Exceptions\FileGenerationAbortedException;
+use Tempest\Generation\Exceptions\FileGenerationFailedException;
+use Tempest\Generation\HasGeneratorCommand;
 use function Tempest\Support\str;
+use Throwable;
 
 trait PublishesFiles
 {
-    use HasConsole;
+    use HasGeneratorCommand;
 
     private array $publishedFiles = [];
 
     private array $publishedClasses = [];
 
     /**
-     * @param Closure(string $source, string $destination): void|null $callback
+     * Publishes a file from a source to a destination.
+     *
+     * @param string $source The path to the source file.
+     * @param string $destination The path to the destination file.
+     * @param Closure(string $source, string $destination): void|null $callback A callback to run after the file is published.
      */
     public function publish(
         string $source,
         string $destination,
         ?Closure $callback = null,
     ): void {
-        if (file_exists($destination)) {
-            if (! $this->confirm(
-                question: "{$destination} already exists Do you want to overwrite it?",
-            )) {
-                return;
-            }
-        } else {
-            if (! $this->confirm(
-                question: "Do you want to create {$destination}?",
+        try {
+            if (! $this->console->confirm(
+                question: sprintf('Do you want to create "%s"', $destination),
                 default: true,
             )) {
-                $this->writeln('Skipped');
+                throw new FileGenerationAbortedException('Skipped.');
```

Comment:
Is this a good idea though? If we don't confirm one file, then the whole installation stop working. From the point of view of installers, that's not how it should work 🤔

Reply:
This is catched at the end of the method reusing the exception message.
So there is no execution break for now 😄

Reply:
Gotcha! I understand now. I don't think I have a very strong opinion on it  for now